tools/chainsync-loadtest/src/fetch_chain.rs

```diff
@@ -0,0 +1,68 @@
+use std::sync::Arc;
+
+use log::info;
+
+use crate::concurrency::{Ctx, Scope};
+use crate::network;
+use anyhow::Context;
+use tokio::time;
+
+use near_primitives::hash::CryptoHash;
+
+pub async fn run(
+    ctx: Ctx,
+    network: Arc<network::Network>,
+    start_block_hash: CryptoHash,
+) -> anyhow::Result<()> {
+    info!("SYNC start");
+    let peers = network.info(&ctx).await?;
+    let target_height = peers.highest_height_peers[0].chain_info.height;
+    info!("SYNC target_height = {}", target_height);
+    Scope::run(&ctx, |ctx, s| async move {
+        s.spawn_weak({
+            let network = network.clone();
+            |ctx| async move {
+                let ctx = ctx.with_label("stats");
+                loop {
+                    info!("stats = {:?}", network.stats);
+                    ctx.wait(time::Duration::from_secs(2)).await?;
+                }
+            }
+        });
+
+        let mut last_hash = start_block_hash;
+        let mut last_height = 0;
+        while last_height < target_height {
+            let mut headers = network.fetch_block_headers(&ctx, &last_hash).await?;
```

Comment:
comment that this is fetching 512 headers after last_hash

Reply:
Done. Added a comment on the fech_block_headers() function.